### PR TITLE
Resolve command format bugs & Add MarkUp unit tests

### LINKED ISSUE #146: [PE-D] Error reported for valid command
The input 'add n/Calculus shlv/textbooks p/41 s/0 q/1 r/free' matches the format 'add n/NAME shlv/SHELF_NAME p/PURCHASE_PRICE s/SELLING_PRICE q/QUANTITY [r/REMARKS]' given in UG but got the response saying the add command format is incorrect.

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: Mist0919/ped#3

### LINKED ISSUE #148: [PE-D] Add Command won't work if the title of the item has a "/"
-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: avellinwong01/ped#15

### LINKED ISSUE #151: [PE-D] Message shown for report command cannot be understood
When a future year-month is keyed in, the error message shown is not very understandable to users. Same with a past year-month

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: avellinwong01/ped#17

### LINKED ISSUE #174: Edit Parser error Message for all commands
Error message should provide more information for the user to debug to check why the invalid commands is not correct such as giving a checking list 

* check unicode char only, no other char

### LINKED ISSUE #182: Restrict adding items with overly long price
Limit for price: 9999.99 (thousands, and 2.dp)
To prevent distortion of list and very possibly a wrong entry

### LINKED ISSUE #189: report date range bugs
image
image

### PR DESCRIPTION
**Pull request by t-l-xin**
Resolve #189 
Resolve #182 
Resolve #151 
Resolve #148 
Resolve #146 
Resolve #174

### CODE REVIEW

**Review by haoyusimon — APPROVED** [review]
Looks good to merge.

File: src/main/java/seedu/duke/parser/Parser.java [hunk 1]
```diff
@@ -33,23 +33,23 @@
...
 
     public static final Pattern ADD_ITEM_DATA_ARGS_FORMAT =
-            Pattern.compile("n/(?<itemName>[^/]+)"
-                    + " shlv/(?<shelfName>[^/]+)"
-                    + " p/(?<purchaseCost>([0-9]+([.][0-9]{1,2})?))"
-                    //only accepts numbers or decimals in 1 or 2 d.p.
-                    + " s/(?<sellingPrice>([0-9]+([.][0-9]{1,2})?))"
-                    //only accepts numbers or decimals in 1 or 2 d.p.
-                    + " q/(?<quantity>[0-9]+)" // only accepts integers, no decimals
-                    + "( r/(?<remarks>[^/]+))?$"); // optional argument
```
Comment (yuejunfeng0909) [inline src/main/java/seedu/duke/parser/Parser.java:43]: https://se-education.org/guides/conventions/java/basic.html#layout

Indentation for wrapped lines should be 8 spaces (i.e. twice the normal indentation of 4 spaces) more than the parent line.

File: text-ui-test/input.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+bye
```
Comment (yuejunfeng0909) [inline text-ui-test/input.txt:1]: Can add back the input?

**Review by yuejunfeng0909 — CHANGES_REQUESTED** [review]

File: text-ui-test/input.txt (hunk 2 above)
Comment (t-l-xin) [reply text-ui-test/input.txt]: okai

File: src/main/java/seedu/duke/parser/Parser.java (hunk 1 above)
Comment (t-l-xin) [reply src/main/java/seedu/duke/parser/Parser.java]: noted

File: src/main/java/seedu/duke/parser/Parser.java [hunk 3]
```diff
@@ -73,13 +73,26 @@
...
+            + "3. For Report Command that require date input ym/YEAR-MONTH should follow input ym/YYYY-MM\n"
+            + "E.g. Jan 2021 is represented as ym/2021-01.\n"
+            + "4. SHELF_NAME & NAME parameters cannot contain special characters e.g.!@#$%^&*[]{}+=`~<>?,./|\\\n"
+            + "5. Do not add extra parameters to commands unless specified in user guide.\n"
+            + "6. Do not leave trailing spaces at the end of the command.\n"
+            + "7. Program input is case-sensitive, and parameter input sequence cannot be reordered.\n"
+            + "8. Program only takes in input from english character set (US-ASCII), "
+            + "no input of other languages are accepted.\n"
+            + "9. INDEX always starts from 1\n"
+            + "For more details refer to User Guide, under Usage Section\n";
```
Comment (yuejunfeng0909) [inline src/main/java/seedu/duke/parser/Parser.java:95]: According to TA, this type of message should be handled by UI, maybe can add it to the predefined message class?

**Review by yuejunfeng0909 — CHANGES_REQUESTED** [review]

**Review by zh1huang — APPROVED** [review]
Looks good to merge

File: src/main/java/seedu/duke/parser/Parser.java (hunk 3 above)
Comment (t-l-xin) [reply src/main/java/seedu/duke/parser/Parser.java:95]: Can

File: src/main/java/seedu/duke/parser/Parser.java [hunk 4]
```diff
@@ -354,9 +354,13 @@ private Command prepareBye(String arguments) throws IllegalFormatException {
      * Parses report command arguments.
      *
      * @param arguments The additional arguments after command word.
+<<<<<<< HEAD
+     * @return Report object
+=======
      *
      * @return TotalCostAndIncomeCommand object
      *
+>>>>>>> master
```
Comment (yuejunfeng0909) [inline src/main/java/seedu/duke/parser/Parser.java:363]: Delete accordingly

**Review by yuejunfeng0909 — CHANGES_REQUESTED** [review]
Generally looks good, just some minor merge error need to resolve

File: src/main/java/seedu/duke/parser/Parser.java (hunk 4 above)
Comment (t-l-xin) [reply src/main/java/seedu/duke/parser/Parser.java:363]: Thanks for pointing this out, missed this cos wasn't returning any error

**Pull request merged**